lib/functions/modem.js

```diff
@@ -353,10 +413,10 @@ module.exports = function (SerialPort) {
       newparts = part.split('\n')
       newparts.forEach(newpart => {
         let pduTest = /[0-9A-Fa-f]{15}/g
-        if (newpart.substr(0, 6) == '+CMTI:') { // New Message Indicatpr with SIM Card ID, After Recieving Read The Message From the SIM Card
+        if (newpart.substr(0, 6) == '+CMTI:') { // New Message Indicator with SIM Card ID, After Recieving Read The Message From the SIM Card
           newpart = newpart.split(',')
```

Comment:
I have question. Here `newpart` is converting from a `String` to an `Array`. Actually it caused me a bug sometimes on line 448 ( `newpart.trim` is not a function, it is `Array` ). What do you suggest?

Reply:
@karianpour CMTI should look like this `+CMTI: "[Message Storage]",[Index]`. On what occasion does it occur? Can you try logging the newpart before the `newpart.split`

Reply:
Here is the `console.log(newpart)` output :
Array(2) ["+CMTI: "SM"", "11"]
modem.js:449
length:2
__proto__:Array(0) [, …]
0:"+CMTI: "SM""
1:"11"
(node:2937) UnhandledPromiseRejectionWarning: TypeError: newpart.trim is not a function
warning.js:18
    at newparts.forEach.newpart (/home/kayvan/heechestan/scraper/node_modules/serialport-gsm/lib/functions/modem.js:451:25)

Reply:
@karianpour  How about putting it like this
```js
let simIndex = newpart.split(',')[1]
modem.readSMSById(simIndex , res => { })
```
can you try it if it works? I don't know why it's getting an error there since newpart is only accessed on that block.